src/hooks/__tests__/useWebSocket.test.ts

```diff
@@ -0,0 +1,106 @@
+import WebSocketServer from 'jest-websocket-mock';
+import { act, renderHook } from '@testing-library/react-hooks';
+
+import useWebSocket from '../useWebSocket';
+
+describe('useWebSocket', () => {
+  it('returns no error when initialized', async () => {
+    const wsServer = new WebSocketServer('ws://localhost/ws/foo');
+    const { result } = renderHook(() => useWebSocket('/foo', jest.fn()));
+
+    // error should be `null` while connecting
+    let [error] = result.current;
+    expect(error).toBeNull();
+
+    await wsServer.connected;
+
+    // error should still be `null` after connection was successful
+    [error] = result.current;
+    expect(error).toBeNull();
+
+    WebSocketServer.clean();
+  });
+
+  it.each([
```

Comment:
`each()` can be used to run the same test with different data sets, e.g. to test different scenarios. Each of the values provided here is passed as an argument to the test (in this case as `data` in line 30).